Bug: https://bugzilla.gnome.org/show_bug.cgi?id=602664
Author: Josselin Mouette <joss@debian.org>

The libgtop_server2 binary is launched by the library itself, and has no use
being launched directly. Therefore, it should go in $(libexecdir) instead.

The patch also adds some missing $(DESTDIR) references in the installation
hooks.

Index: libgtop-2.32.0/libgtop-sysdeps.m4
===================================================================
--- libgtop-2.32.0.orig/libgtop-sysdeps.m4	2015-10-07 23:46:35.738589058 +0200
+++ libgtop-2.32.0/libgtop-sysdeps.m4	2015-10-07 23:51:51.006990769 +0200
@@ -40,38 +40,38 @@
 	netbsd*|bsdi*)
 	  libgtop_sysdeps_dir=bsd
 	  libgtop_need_server=yes
-	  libgtop_postinstall='chgrp kmem $(bindir)/libgtop_server2 && chmod 2755 $(bindir)/libgtop_server2'
+	  libgtop_postinstall='chgrp kmem $(DESTDIR)$(libexecdir)/libgtop_server2 && chmod 2755 $(DESTDIR)$(libexecdir)/libgtop_server2'
 	  ;;
 	openbsd*)
 	  libgtop_sysdeps_dir=openbsd
 	  libgtop_need_server=yes
 	  libgtop_sysdeps_private_mountlist=yes
 	  libgtop_sysdeps_private_fsusage=yes
-	  libgtop_postinstall='chgrp kmem $(bindir)/libgtop_server2 && chmod 2555 $(bindir)/libgtop_server2'
+	  libgtop_postinstall='chgrp kmem $(DESTDIR)$(libexecdir)/libgtop_server2 && chmod 2755 $(DESTDIR)$(libexecdir)/libgtop_server2'
 	  ;;
 	freebsd*|kfreebsd*)
 	  libgtop_sysdeps_dir=freebsd
 	  libgtop_need_server=yes
 	  libgtop_sysdeps_private_mountlist=yes
 	  libgtop_sysdeps_private_fsusage=yes
-	  libgtop_postinstall='chgrp kmem $(bindir)/libgtop_server2 && chmod 2755 $(bindir)/libgtop_server2'
+	  libgtop_postinstall='chgrp kmem $(DESTDIR)$(libexecdir)/libgtop_server2 && chmod 2755 $(DESTDIR)$(libexecdir)/libgtop_server2'
 	  ;;
 	solaris*)
 	  libgtop_sysdeps_dir=solaris
 	  libgtop_need_server=yes
-	  libgtop_postinstall='chgrp sys $(bindir)/libgtop_server && chmod 2755 $(bindir)/libgtop_server'
+	  libgtop_postinstall='chgrp sys $(DESTDIR)$(libexecdir)/libgtop_server && chmod 2755 $(DESTDIR)$(libexecdir)/libgtop_server'
 	  ;;
 	aix*)
 	  libgtop_sysdeps_dir=aix
 	  libgtop_need_server=yes
 	  libgtop_have_sysinfo=yes
-	  libgtop_postinstall='chgrp system $(bindir)/libgtop_server && chmod g+s $(bindir)/libgtop_server2'
+	  libgtop_postinstall='chgrp system $(DESTDIR)$(libexecdir)/libgtop_server && chmod g+s $(DESTDIR)$(libexecdir)/libgtop_server2'
 	  ;;
 	darwin*)
 	  libgtop_sysdeps_dir=darwin
 	  libgtop_need_server=yes
 	  libgtop_have_sysinfo=yes
-	  libgtop_postinstall='chgrp kmem $(bindir)/libgtop_server2 && chmod g+s $(bindir)/libgtop_server2'
+	  libgtop_postinstall='chgrp kmem $(DESTDIR)$(libexecdir)/libgtop_server2 && chmod g+s $(DESTDIR)$(libexecdir)/libgtop_server2'
 	  ;;
 	cygwin*)
 	  libgtop_sysdeps_dir=cygwin
Index: libgtop-2.32.0/src/daemon/Makefile.am
===================================================================
--- libgtop-2.32.0.orig/src/daemon/Makefile.am	2015-10-07 23:46:35.738589058 +0200
+++ libgtop-2.32.0/src/daemon/Makefile.am	2015-10-07 23:46:35.734589003 +0200
@@ -26,7 +26,7 @@
 suid_common			=
 endif
 
-bin_PROGRAMS			= libgtop_daemon2 @server_programs@
+libexec_PROGRAMS		= libgtop_daemon2 @server_programs@
 
 EXTRA_PROGRAMS			= libgtop_server2
 
Index: libgtop-2.32.0/configure.ac
===================================================================
--- libgtop-2.32.0.orig/configure.ac	2015-10-07 23:46:35.738589058 +0200
+++ libgtop-2.32.0/configure.ac	2015-10-07 23:46:35.734589003 +0200
@@ -285,8 +285,8 @@
 libgtop_save_exec_prefix="$exec_prefix"
 test "x$prefix" = xNONE && prefix=$ac_default_prefix
 test "x$exec_prefix" = xNONE && exec_prefix=$prefix
-LIBGTOP_BINDIR=`eval echo "${bindir}"`
-LIBGTOP_SERVER=`eval echo "${bindir}/libgtop_server2"`
+LIBGTOP_BINDIR=`eval echo "${libexecdir}"`
+LIBGTOP_SERVER=`eval echo "${libexecdir}/libgtop_server2"`
 prefix="$libgtop_save_prefix"
 exec_prefix="$libgtop_save_exec_prefix"
 
